modulemd/v2/tests/test_data/dependson_v2.yaml

```diff
@@ -14,8 +14,10 @@ data:
   dependencies:
   - buildrequires:
       platform: [ f30 ]
+      streamname: [] 
```

Comment:
You added this field but didn't add a test for `stream.build_depends_on_stream('streamname', 'f30', True)

Reply:
self.assertEqual(
                stream.build_depends_on_stream(
                    'streamname', 'f30'), False)
ninja test is failing when I give this test.
It fails when I give True also. What should be the condition to check with? Is it like it could be Either true or false?

Reply:
The empty set ( `[]` ) should mean that any stream name should be accepted. It's possible that you have found a bug in the code (which is the purpose of writing new tests!).

The expected behavior is that `stream.build_depends_on_stream('streamname', 'f30'), True)` should be successful here. If it's not, we should look into the code this is exercising and try to figure out where it's going wrong.

Reply:
Is this the file to be modified to fix that bug:
[link](https://github.com/fedora-modularity/libmodulemd/blob/4e60ff2b1194b7da8648517411d48a4f4b7ae471/modulemd/v2/modulemd-module-stream-v2.c) 
Can you give me a lead to fix it.

Reply:
@athira-selvam That is indeed the correct location, but I don't see anything obviously wrong with the code. Please push your work-in-progress to this PR and I'll see if there's a mistake in your test.

Reply:
I  have pushed my work-in-progress.

Reply:
I found the problem: you are running this test on both the `Modulemd.ModuleStreamV1` and `Modulemd.ModuleStreamV2` objects.

There is no `[ ]` implementation for V1, so you can't use this test there. Wrap this test in a check to make sure it only runs on V2 or later:

```
if (version >= Modulemd.ModuleStreamVersionEnum.TWO):
```